### ISSUE TITLE
Detectable Game Objects not working with 2D gameobjects

### ISSUE THREAD

**Issue by SandaruwanFdo** [mbaske/grid-sensor#14]
![image](https://user-images.githubusercontent.com/5012179/187504333-b2acf41d-6add-4ae6-a246-4d999be7df41.png)
![image](https://user-images.githubusercontent.com/5012179/187504472-306da6c5-0902-4f53-aafa-91b4ee7be22f.png)

Detectable Game Objects not working with 2D gameobjects. what am I missing?

**Comment by mbaske** [comment]
> Detectable Game Objects not working with 2D gameobjects. what am I missing?

Yeah, sorry,, that's expected behavior though. Please see readme page:
_"I debated with myself if I should add support for 2D physics, but decided against it in the end. The reason being that the typical use case for 2D physics is platformer games. Those tend to require pretty high resolution observations though. Using camera sensors should generally be the better option for platformers. "_

**Comment by SandaruwanFdo** [comment]
I read it now. Do you have any solution for Top Down games?

**Comment by mbaske** [comment]
You could either use 3D physics or implement the detection code yourself - that is mapping your object positions to grid coordinates, like I did with the "maze explorer" example project in the repo.

**Comment by SandaruwanFdo** [comment]
Thank you for your response. Do you intend to provide a solution for that in your sensor in the future?

**Comment by mbaske** [comment]
No, I have nothing planned for the time being.

**Comment by SandaruwanFdo** [comment]
> **object positions to grid coordinates**
Can you provide a some examples or a video tutorial?  "maze explorer" is a bit complex for me. I am a stranger to ml-agents

**Comment by mbaske** [comment]
Hi, I don't have any docs besides what's in the repo's readme. If you're totally new to ml-agents, I'd suggest looking into the example projects first. They are pretty basic and helped me a lot when I was starting with the ml-agents.
https://github.com/Unity-Technologies/ml-agents/tree/main/Project/Assets/ML-Agents/Examples
